File: Trunk/vpPriV100GrupoMundifios/Inditex/Base/FichaFornecedor/BasIsFichaFornecedor.cs
using Generico;
using Microsoft.VisualBasic;
using Primavera.Extensibility.Base.Editors;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs;
using Primavera.Extensibility.Constants.ExtensibilityService;
using System.Windows.Forms;

namespace Inditex
{
    public class BasIsFichaFornecedor : FichaFornecedores
    {
        public override void TeclaPressionada(int KeyCode, int Shift, ExtensibilityEventArgs e)
        {
            base.TeclaPressionada(KeyCode, Shift, e);

            if (Module1.VerificaToken("Inditex") == 1)
            {
                // Bruno Peixoto 02/09/2020 - CTRL+O para abrir o formumlario de Fiações Inditex
                if (KeyCode == 79 & this.Fornecedor.Inactivo == false & (Strings.UCase(Aplicacao.Utilizador.Utilizador) == "ANA" | Strings.UCase(Aplicacao.Utilizador.Utilizador) == "RICARDO" | Strings.UCase(Aplicacao.Utilizador.Utilizador) == "SUPORTE" | Strings.UCase(Aplicacao.Utilizador.Utilizador) == "INFORMATICA"))
                {
                    Module1.certFiacoes = this.Fornecedor.Fornecedor;

                    ExtensibilityResult result = BSO.Extensibility.CreateCustomFormInstance(typeof(FrmInditex));

                    if (result.ResultCode == ExtensibilityResultCode.Ok)
                    {
                        FrmInditex frm = result.Result;
                        frm.ShowDialog();
                    }
                }

                if (KeyCode == 79 & this.Fornecedor.Inactivo == true)
                    MessageBox.Show("Fornecedor Anulado! Não é possível abrir o formulário de Fiações Inditex!", "", MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
            }
        }
    }
}

File: Branches/VBInicial/vpPriV100GrupoMundifios/Inditex/Base/FichaFornecedor/BasIsFichaFornecedor.vb
Imports Primavera.Extensibility.Base.Editors
Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs
Imports StdBE100
Imports Vimaponto.PrimaveraV100.Clientes.GrupoMundifios.Generico

Namespace Inditex
    Public Class BasIsFichaFornecedor
        Inherits FichaFornecedores

        'Public Overrides Sub TeclaPressionada(KeyCode As Integer, Shift As Integer, e As ExtensibilityEventArgs)
        '    MyBase.TeclaPressionada(KeyCode, Shift, e)

        '    If Module1.VerificaToken("Inditex") = 1 Then

        '        'Bruno Peixoto 02/09/2020 - CTRL+O para abrir o formumlario de Fiações Inditex
        '        If KeyCode = 79 And Me.Fornecedor.Inactivo = False And (UCase(Aplicacao.Utilizador.Utilizador) = "ANA" Or UCase(Aplicacao.Utilizador.Utilizador) = "RICARDO" Or UCase(Aplicacao.Utilizador.Utilizador) = "SUPORTE" Or UCase(Aplicacao.Utilizador.Utilizador) = "INFORMATICA") Then
        '            Module1.certFiacoes = Me.Fornecedor.Fornecedor

        '            Load FrmInditexFornecedor
        '    FrmInditexFornecedor.Show
        '        End If

        '        If KeyCode = 79 And Me.Fornecedor.Inactivo = True Then
        '            MsgBox("Fornecedor Anulado! Não é possível abrir o formulário de Fiações Inditex!", vbInformation)
        '        End If

        '    End If

        'End Sub

    End Class
End Namespace
